app.js

```diff
@@ -56,30 +61,39 @@ app.use(cookieParser("use_a_secure_secret"))
 
 // CSRF middleware with __Host- cookie
 const csrfProtection = csrf({
-    cookie: {
-      key: "csrftoken",
-      httpOnly: true,
-      secure: app.get("env") === "production", // require HTTPS in prod
-      sameSite: "strict",
-      path: "/", // required for __Host-
-    },
-  });
-
-  app.use(csrfProtection);
-
-  // Makes token available to all views
-  app.use((req, res, next) => {
-    res.locals.csrfToken = req.csrfToken();
-    next();
-  });
+  cookie: {
```

Comment:
This is not best practice.  You are using a session.  You should not use a separate cookie for the csrf if you are using a session.  You should configure csrf with cookie: false.  You would then not need the cookie parser middleware.  Also, you have configured the cookie parser with the secret "use_a_secure_secret".  This is for encrypted cookies.  You are not using encrypted cookies, but in any case, you should never put a secret into your code!  You should load it from the .env.